layouts/partials/head.html

```diff
@@ -14,9 +16,29 @@
     <meta property="og:description" content="{{ .Description }}">
     <meta property="og:url" content="{{ .Permalink }}">
     <meta property="og:site_name" content="{{ .Site.Title }}">
+    {{ if .Params.thumbnail }}
+    <meta property="og:image" content="{{ .Site.BaseURL }}{{ .Params.thumbnail }}">
+    {{ end }}
     {{ range .Params.tags }}
     <meta property="og:tags" content="{{ . }}">
     {{ end }}
+    <!-- End of Open Graph -->
+    <!-- Start of Twitter -->
+    {{ if .Site.Params.twitter_card_whitelisted }}
```

Comment:
`twitter_card_whitelisted` is needed because by default Twitter doesn't whitelist the field `<meta property="twitter:card">` but it's not vital either.